File: recipe-site/app/_models/FileMetadata.ts
import { FileRole } from "../_constants/FileRole";

export default class FileMetadata {
    fileId: number;
    uuId: string;
    filename: string;
    contentType: string;
    sizeBytes: number;
    fileRole: FileRole;
    url: string | undefined;

    constructor(
        fileId: number,
        uuId: string,
        filename: string,
        contentType: string,
        sizeBytes: number,
        fileRole: FileRole,
        url?: string
    ) {
        this.fileId = fileId;
        this.uuId = uuId;
        this.filename = filename;
        this.contentType = contentType;
        this.sizeBytes = sizeBytes;
        this.fileRole = fileRole;
        this.url = url;
    }
}

File: java-recipes/src/main/java/com/fargopolis/models/FileMetadata.java
package com.fargopolis.models;

import com.fargopolis.enums.FileRole;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class FileMetadata {
	private Integer fileId;
	private String uuId;
	private String filename;
	private String contentType;
	private Long sizeBytes;
	private FileRole fileRole;
	private String url;
}
